File: src/github.rs
use anyhow::{Context, Result, anyhow};
use serde::{Deserialize, Serialize};
use std::process::Command;

use crate::config::{Todo, TodoStatus};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GitHubProject {
    pub id: String,
    pub number: u32,
    pub title: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProjectItem {
    pub id: String,
    pub content: ProjectItemContent,
    pub field_values: Vec<FieldValue>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProjectItemContent {
    pub title: String,
    pub body: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FieldValue {
    pub field: String,
    pub value: String,
}

pub struct GitHubClient {
    owner: String,
    repo: String,
    project_number: u32,
}

impl GitHubClient {
    pub fn new(owner: String, repo: String, project_number: u32) -> Self {
        Self {
            owner,
            repo,
            project_number,
        }
    }

    /// Check if gh CLI is authenticated
    pub fn is_authenticated() -> Result<bool> {
        let output = Command::new("gh")
            .args(&["auth", "status"])
            .output()
            .context("Failed to check gh auth status")?;

        Ok(output.status.success())
    }

    /// List available projects for the repository
    pub fn list_projects(owner: &str, repo: &str) -> Result<Vec<GitHubProject>> {
        let query = format!(
            r#"
            query {{
              repository(owner: "{}", name: "{}") {{
                projectsV2(first: 10) {{
                  nodes {{
                    id
                    number
                    title
                  }}
                }}
              }}
            }}
            "#,
            owner, repo
        );

        let output = Command::new("gh")
            .args(&["api", "graphql", "-f", &format!("query={}", query)])
            .output()
            .context("Failed to list projects")?;

        if !output.status.success() {
            let stderr = String::from_utf8_lossy(&output.stderr);
            return Err(anyhow!("Failed to list projects: {}", stderr));
        }

        let response: serde_json::Value = serde_json::from_slice(&output.stdout)
            .context("Failed to parse project list response")?;

        let projects: Vec<GitHubProject> = response["data"]["repository"]["projectsV2"]["nodes"]
            .as_array()
            .context("Invalid project list format")?
            .iter()
            .filter_map(|node| {
                Some(GitHubProject {
                    id: node["id"].as_str()?.to_string(),
                    number: node["number"].as_u64()? as u32,
                    title: node["title"].as_str()?.to_string(),
                })
            })
            .collect();

        Ok(projects)
    }

    /// Get project ID from project number
    fn get_project_id(&self) -> Result<String> {
        let query = format!(
            r#"
            query {{
              repository(owner: "{}", name: "{}") {{
                projectsV2(first: 10) {{
                  nodes {{
                    id
                    number
                  }}
                }}
              }}
            }}
            "#,
            self.owner, self.repo
        );

        let output = Command::new("gh")
            .args(&["api", "graphql", "-f", &format!("query={}", query)])
            .output()
            .context("Failed to get project ID")?;

        if !output.status.success() {
            let stderr = String::from_utf8_lossy(&output.stderr);
            return Err(anyhow!("Failed to get project ID: {}", stderr));
        }

        let response: serde_json::Value = serde_json::from_slice(&output.stdout)
            .context("Failed to parse project ID response")?;

        let nodes = response["data"]["repository"]["projectsV2"]["nodes"]
            .as_array()
            .context("Invalid project list format")?;

        for node in nodes {
            if node["number"].as_u64() == Some(self.project_number as u64) {
                return Ok(node["id"]
                    .as_str()
                    .context("Missing project ID")?
                    .to_string());
            }
        }

        Err(anyhow!("Project {} not found", self.project_number))
    }

    /// Fetch all todos from GitHub Project
    pub fn fetch_todos(&self) -> Result<Vec<Todo>> {
        let project_id = self.get_project_id()?;

        let query = format!(
            r#"
            query {{
              node(id: "{}") {{
                ... on ProjectV2 {{
                  items(first: 100) {{
                    nodes {{
                      id
                      content {{
                        ... on Issue {{
                          title
                          body
                        }}
                        ... on DraftIssue {{
                          title
                          body
                        }}
                      }}
                      fieldValues(first: 10) {{
                        nodes {{
                          ... on ProjectV2ItemFieldSingleSelectValue {{
                            field {{
                              ... on ProjectV2SingleSelectField {{
                                name
                              }}
                            }}
                            name
                          }}
                          ... on ProjectV2ItemFieldTextValue {{
                            field {{
                              ... on ProjectV2Field {{
                                name
                              }}
                            }}
                            text
                          }}
                        }}
                      }}
                    }}
                  }}
                }}
              }}
            }}
            "#,
            project_id
        );

        let output = Command::new("gh")
            .args(&["api", "graphql", "-f", &format!("query={}", query)])
            .output()
            .context("Failed to fetch project items")?;

        if !output.status.success() {
            let stderr = String::from_utf8_lossy(&output.stderr);
            return Err(anyhow!("Failed to fetch project items: {}", stderr));
        }

        let response: serde_json::Value = serde_json::from_slice(&output.stdout)
            .context("Failed to parse project items response")?;

        let items = response["data"]["node"]["items"]["nodes"]
            .as_array()
            .context("Invalid project items format")?;

        let mut todos = Vec::new();

        for item in items {
            let title = item["content"]["title"]
                .as_str()
                .unwrap_or("Untitled")
                .to_string();

            // Look for status field
            let mut status = TodoStatus::Pending;
            let mut worktree_name: Option<String> = None;

            if let Some(field_values) = item["fieldValues"]["nodes"].as_array() {
                for field_value in field_values {
                    if let Some(field_name) = field_value["field"]["name"].as_str() {
                        if field_name == "Status" {
                            if let Some(value) = field_value["name"].as_str() {
                                if value.to_lowercase().contains("done")
                                    || value.to_lowercase().contains("complete")
                                {
                                    status = TodoStatus::Done;
                                }
                            }
                        } else if field_name == "Worktree" {
                            worktree_name = field_value["text"].as_str().map(|s| s.to_string());
                        }
                    }
                }
            }

            todos.push(Todo {
                description: title,
                status,
                worktree: worktree_name,
            });
        }

        Ok(todos)
    }

    /// Add a new todo to GitHub Project
    pub fn add_todo(&self, description: &str, _worktree_name: &str) -> Result<()> {
        let project_id = self.get_project_id()?;

        // First, create a draft issue in the project
        let mutation = format!(
            r#"
            mutation {{
              addProjectV2DraftIssue(input: {{
                projectId: "{}"
                title: "{}"
              }}) {{
                projectItem {{
                  id
                }}
              }}
            }}
            "#,
            project_id,
            description.replace('"', "\\\"")
        );

        let output = Command::new("gh")
            .args(&["api", "graphql", "-f", &format!("query={}", mutation)])
            .output()
            .context("Failed to add project item")?;

        if !output.status.success() {
            let stderr = String::from_utf8_lossy(&output.stderr);
            return Err(anyhow!("Failed to add project item: {}", stderr));
        }

        // TODO: Set the worktree field value
        // This would require getting the field ID first, then updating the item
        // For now, we'll just create the item

        Ok(())
    }

    /// Mark a todo as done in GitHub Project
    pub fn mark_todo_done(&self, _worktree_name: &str) -> Result<()> {
        // TODO: Implement marking todo as done
        // This would require:
        // 1. Finding the item by worktree name
        // 2. Getting the Status field ID
        // 3. Getting the "Done" option ID
        // 4. Updating the item's status field
        Ok(())
    }

    /// Sync todos from GitHub to local cache
    pub fn sync_from_github(&self) -> Result<Vec<Todo>> {
        self.fetch_todos()
    }

    /// Sync todos from local cache to GitHub
    #[allow(dead_code)]
    pub fn sync_to_github(&self, _todos: &[Todo]) -> Result<()> {
        // TODO: Implement bidirectional sync
        // This would require:
        // 1. Comparing local vs remote todos
        // 2. Creating new items for local todos not in GitHub
        // 3. Updating status for changed items
        // 4. Optionally deleting items removed locally
        Ok(())
    }
}

/// Get current repository owner and name from git remote
pub fn get_repo_info() -> Result<(String, String)> {
    let output = Command::new("gh")
        .args(&["repo", "view", "--json", "owner,name"])
        .output()
        .context("Failed to get repository info")?;

    if !output.status.success() {
        let stderr = String::from_utf8_lossy(&output.stderr);
        return Err(anyhow!("Failed to get repository info: {}", stderr));
    }

    let response: serde_json::Value = serde_json::from_slice(&output.stdout)
        .context("Failed to parse repository info")?;

    let owner = response["owner"]["login"]
        .as_str()
        .context("Missing owner")?
        .to_string();

    let name = response["name"]
        .as_str()
        .context("Missing repo name")?
        .to_string();

    Ok((owner, name))
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_github_client_creation() {
        let client = GitHubClient::new(
            "owner".to_string(),
            "repo".to_string(),
            1,
        );
        assert_eq!(client.owner, "owner");
        assert_eq!(client.repo, "repo");
        assert_eq!(client.project_number, 1);
    }

    #[test]
    fn test_todo_status_parsing() {
        let todo = Todo {
            description: "Test".to_string(),
            status: TodoStatus::Pending,
            worktree: Some("test-worktree".to_string()),
        };
        assert_eq!(todo.status, TodoStatus::Pending);
    }
}

File: internal/github/github.go
package github

import (
	"bytes"
	"encoding/json"
	"fmt"
	"os/exec"
	"strings"
)

type Project struct {
	ID     string `json:"id"`
	Number int    `json:"number"`
	Title  string `json:"title"`
}

type ProjectItem struct {
	ID      string `json:"id"`
	Title   string `json:"title"`
	Status  string `json:"status"`
	Body    string `json:"body"`
	Content struct {
		Number int    `json:"number"`
		Title  string `json:"title"`
		Body   string `json:"body"`
		URL    string `json:"url"`
	} `json:"content"`
}

type RepoInfo struct {
	Owner string
	Name  string
}

// IsAuthenticated checks if gh CLI is authenticated
func IsAuthenticated() bool {
	cmd := exec.Command("gh", "auth", "status")
	return cmd.Run() == nil
}

// HasRequiredScopes checks if the token has project and repo scopes
func HasRequiredScopes() (bool, error) {
	cmd := exec.Command("gh", "auth", "status", "-t")
	output, err := cmd.Output()
	if err != nil {
		return false, nil
	}

	// Parse output to check for scopes
	lines := strings.Split(string(output), "\n")
	for _, line := range lines {
		if strings.Contains(line, "Token scopes:") {
			hasProject := strings.Contains(line, "'project'") || strings.Contains(line, "project")
			hasRepo := strings.Contains(line, "'repo'") || strings.Contains(line, "repo")
			return hasProject && hasRepo, nil
		}
	}

	return false, nil
}

// Authenticate triggers GitHub authentication with required scopes
func Authenticate() error {
	cmd := exec.Command("gh", "auth", "refresh", "-h", "github.com", "-s", "project", "-s", "repo")
	cmd.Stdin = nil
	cmd.Stdout = nil
	cmd.Stderr = nil
	return cmd.Run()
}

// GetRepoInfo gets the current repository owner and name
func GetRepoInfo() (*RepoInfo, error) {
	cmd := exec.Command("gh", "repo", "view", "--json", "owner,name")
	output, err := cmd.Output()
	if err != nil {
		return nil, fmt.Errorf("failed to get repo info: %w", err)
	}

	var result struct {
		Owner struct {
			Login string `json:"login"`
		} `json:"owner"`
		Name string `json:"name"`
	}

	if err := json.Unmarshal(output, &result); err != nil {
		return nil, fmt.Errorf("failed to parse repo info: %w", err)
	}

	return &RepoInfo{
		Owner: result.Owner.Login,
		Name:  result.Name,
	}, nil
}

// ListProjects lists all GitHub Projects for a repository
func ListProjects(owner, repo string) ([]Project, error) {
	query := fmt.Sprintf(`
		query {
			repository(owner: "%s", name: "%s") {
				projectsV2(first: 10) {
					nodes {
						id
						number
						title
					}
				}
			}
		}
	`, owner, repo)

	output, err := runGraphQL(query)
	if err != nil {
		return nil, err
	}

	var result struct {
		Data struct {
			Repository struct {
				ProjectsV2 struct {
					Nodes []Project `json:"nodes"`
				} `json:"projectsV2"`
			} `json:"repository"`
		} `json:"data"`
	}

	if err := json.Unmarshal(output, &result); err != nil {
		return nil, fmt.Errorf("failed to parse projects: %w", err)
	}

	return result.Data.Repository.ProjectsV2.Nodes, nil
}

// CreateProject creates a new GitHub Project
func CreateProject(owner, repo, title string) (*Project, error) {
	// Get both repository ID and owner ID
	repoQuery := fmt.Sprintf(`
		query {
			repository(owner: "%s", name: "%s") {
				id
				owner {
					id
				}
			}
		}
	`, owner, repo)

	output, err := runGraphQL(repoQuery)
	if err != nil {
		return nil, err
	}

	var repoResult struct {
		Data struct {
			Repository struct {
				ID    string `json:"id"`
				Owner struct {
					ID string `json:"id"`
				} `json:"owner"`
			} `json:"repository"`
		} `json:"data"`
	}

	if err := json.Unmarshal(output, &repoResult); err != nil {
		return nil, fmt.Errorf("failed to parse repo ID: %w", err)
	}

	ownerID := repoResult.Data.Repository.Owner.ID
	repoID := repoResult.Data.Repository.ID

	// Create the project with the owner ID
	mutation := fmt.Sprintf(`
		mutation {
			createProjectV2(input: {
				ownerId: "%s"
				title: "%s"
			}) {
				projectV2 {
					id
					number
					title
				}
			}
		}
	`, ownerID, escapeString(title))

	output, err = runGraphQL(mutation)
	if err != nil {
		return nil, err
	}

	var createResult struct {
		Data struct {
			CreateProjectV2 struct {
				ProjectV2 Project `json:"projectV2"`
			} `json:"createProjectV2"`
		} `json:"data"`
	}

	if err := json.Unmarshal(output, &createResult); err != nil {
		return nil, fmt.Errorf("failed to parse project creation: %w", err)
	}

	project := createResult.Data.CreateProjectV2.ProjectV2

	// Link the project to the repository
	linkMutation := fmt.Sprintf(`
		mutation {
			linkProjectV2ToRepository(input: {
				projectId: "%s"
				repositoryId: "%s"
			}) {
				repository {
					id
				}
			}
		}
	`, project.ID, repoID)

	_, err = runGraphQL(linkMutation)
	if err != nil {
		// Don't fail if linking fails, project is still created
		fmt.Printf("Warning: failed to link project to repository: %v\n", err)
	}

	return &project, nil
}

func runGraphQL(query string) ([]byte, error) {
	cmd := exec.Command("gh", "api", "graphql", "-f", fmt.Sprintf("query=%s", query))
	var stderr bytes.Buffer
	cmd.Stderr = &stderr

	output, err := cmd.Output()
	if err != nil {
		return nil, fmt.Errorf("GraphQL query failed: %s", stderr.String())
	}

	return output, nil
}

// ListProjectItems fetches all items from a GitHub Project
func ListProjectItems(owner, repo string, projectNumber int) ([]ProjectItem, error) {
	// First, get the project ID
	projectQuery := fmt.Sprintf(`
		query {
			repository(owner: "%s", name: "%s") {
				projectsV2(first: 10) {
					nodes {
						id
						number
						title
					}
				}
			}
		}
	`, owner, repo)

	output, err := runGraphQL(projectQuery)
	if err != nil {
		return nil, err
	}

	var projectResult struct {
		Data struct {
			Repository struct {
				ProjectsV2 struct {
					Nodes []Project `json:"nodes"`
				} `json:"projectsV2"`
			} `json:"repository"`
		} `json:"data"`
	}

	if err := json.Unmarshal(output, &projectResult); err != nil {
		return nil, fmt.Errorf("failed to parse projects: %w", err)
	}

	// Find the project with the matching number
	var projectID string
	for _, project := range projectResult.Data.Repository.ProjectsV2.Nodes {
		if project.Number == projectNumber {
			projectID = project.ID
			break
		}
	}

	if projectID == "" {
		return nil, fmt.Errorf("project #%d not found", projectNumber)
	}

	// Get the project items with status field
	itemsQuery := fmt.Sprintf(`
		query {
			node(id: "%s") {
				... on ProjectV2 {
					items(first: 100) {
						nodes {
							id
							fieldValues(first: 10) {
								nodes {
									... on ProjectV2ItemFieldSingleSelectValue {
										name
										field {
											... on ProjectV2SingleSelectField {
												name
											}
										}
									}
									... on ProjectV2ItemFieldTextValue {
										text
										field {
											... on ProjectV2FieldCommon {
												name
											}
										}
									}
								}
							}
							content {
								... on Issue {
									number
									title
									body
									url
								}
								... on DraftIssue {
									title
									body
								}
							}
						}
					}
				}
			}
		}
	`, projectID)

	output, err = runGraphQL(itemsQuery)
	if err != nil {
		return nil, err
	}

	var itemsResult struct {
		Data struct {
			Node struct {
				Items struct {
					Nodes []struct {
						ID          string `json:"id"`
						FieldValues struct {
							Nodes []struct {
								Name  string `json:"name"`
								Text  string `json:"text"`
								Field struct {
									Name string `json:"name"`
								} `json:"field"`
							} `json:"nodes"`
						} `json:"fieldValues"`
						Content struct {
							Number int    `json:"number"`
							Title  string `json:"title"`
							Body   string `json:"body"`
							URL    string `json:"url"`
						} `json:"content"`
					} `json:"nodes"`
				} `json:"items"`
			} `json:"node"`
		} `json:"data"`
	}

	if err := json.Unmarshal(output, &itemsResult); err != nil {
		return nil, fmt.Errorf("failed to parse project items: %w", err)
	}

	// Convert to ProjectItem
	var items []ProjectItem
	for _, node := range itemsResult.Data.Node.Items.Nodes {
		item := ProjectItem{
			ID:      node.ID,
			Title:   node.Content.Title,
			Content: node.Content,
		}

		// Extract status from field values
		for _, fv := range node.FieldValues.Nodes {
			if fv.Field.Name == "Status" {
				item.Status = fv.Name
				break
			}
		}

		items = append(items, item)
	}

	return items, nil
}

// CreateProjectItem creates a new item in a GitHub Project
func CreateProjectItem(owner, repo string, projectNumber int, title string) (*ProjectItem, error) {
	// First, get the project ID
	projectQuery := fmt.Sprintf(`
		query {
			repository(owner: "%s", name: "%s") {
				projectsV2(first: 10) {
					nodes {
						id
						number
					}
				}
			}
		}
	`, owner, repo)

	output, err := runGraphQL(projectQuery)
	if err != nil {
		return nil, err
	}

	var projectResult struct {
		Data struct {
			Repository struct {
				ProjectsV2 struct {
					Nodes []struct {
						ID     string `json:"id"`
						Number int    `json:"number"`
					} `json:"nodes"`
				} `json:"projectsV2"`
			} `json:"repository"`
		} `json:"data"`
	}

	if err := json.Unmarshal(output, &projectResult); err != nil {
		return nil, fmt.Errorf("failed to parse projects: %w", err)
	}

	var projectID string
	for _, project := range projectResult.Data.Repository.ProjectsV2.Nodes {
		if project.Number == projectNumber {
			projectID = project.ID
			break
		}
	}

	if projectID == "" {
		return nil, fmt.Errorf("project #%d not found", projectNumber)
	}

	// Create a draft issue in the project
	mutation := fmt.Sprintf(`
		mutation {
			addProjectV2DraftIssue(input: {
				projectId: "%s"
				title: "%s"
			}) {
				projectItem {
					id
					content {
						... on DraftIssue {
							title
						}
					}
				}
			}
		}
	`, projectID, escapeString(title))

	output, err = runGraphQL(mutation)
	if err != nil {
		return nil, fmt.Errorf("failed to create project item: %w", err)
	}

	var createResult struct {
		Data struct {
			AddProjectV2DraftIssue struct {
				ProjectItem struct {
					ID      string `json:"id"`
					Content struct {
						Title string `json:"title"`
					} `json:"content"`
				} `json:"projectItem"`
			} `json:"addProjectV2DraftIssue"`
		} `json:"data"`
	}

	if err := json.Unmarshal(output, &createResult); err != nil {
		return nil, fmt.Errorf("failed to parse project item creation: %w", err)
	}

	return &ProjectItem{
		ID:    createResult.Data.AddProjectV2DraftIssue.ProjectItem.ID,
		Title: createResult.Data.AddProjectV2DraftIssue.ProjectItem.Content.Title,
	}, nil
}

// UpdateProjectItemStatus updates the status of a project item
func UpdateProjectItemStatus(owner, repo string, projectNumber int, itemID string, status string) error {
	// First, get the project ID and status field ID
	projectQuery := fmt.Sprintf(`
		query {
			repository(owner: "%s", name: "%s") {
				projectsV2(first: 10) {
					nodes {
						id
						number
						fields(first: 20) {
							nodes {
								... on ProjectV2SingleSelectField {
									id
									name
									options {
										id
										name
									}
								}
							}
						}
					}
				}
			}
		}
	`, owner, repo)

	output, err := runGraphQL(projectQuery)
	if err != nil {
		return err
	}

	var projectResult struct {
		Data struct {
			Repository struct {
				ProjectsV2 struct {
					Nodes []struct {
						ID     string `json:"id"`
						Number int    `json:"number"`
						Fields struct {
							Nodes []struct {
								ID      string `json:"id"`
								Name    string `json:"name"`
								Options []struct {
									ID   string `json:"id"`
									Name string `json:"name"`
								} `json:"options"`
							} `json:"nodes"`
						} `json:"fields"`
					} `json:"nodes"`
				} `json:"projectsV2"`
			} `json:"repository"`
		} `json:"data"`
	}

	if err := json.Unmarshal(output, &projectResult); err != nil {
		return fmt.Errorf("failed to parse projects: %w", err)
	}

	// Find the project and status field
	var projectID, statusFieldID, statusOptionID string
	for _, project := range projectResult.Data.Repository.ProjectsV2.Nodes {
		if project.Number == projectNumber {
			projectID = project.ID
			// Find the Status field
			for _, field := range project.Fields.Nodes {
				if field.Name == "Status" {
					statusFieldID = field.ID
					// Find the option matching the desired status
					for _, option := range field.Options {
						if option.Name == status {
							statusOptionID = option.ID
							break
						}
					}
					break
				}
			}
			break
		}
	}

	if projectID == "" {
		return fmt.Errorf("project #%d not found", projectNumber)
	}
	if statusFieldID == "" {
		return fmt.Errorf("Status field not found in project")
	}
	if statusOptionID == "" {
		return fmt.Errorf("status option '%s' not found", status)
	}

	// Update the item status
	mutation := fmt.Sprintf(`
		mutation {
			updateProjectV2ItemFieldValue(input: {
				projectId: "%s"
				itemId: "%s"
				fieldId: "%s"
				value: {
					singleSelectOptionId: "%s"
				}
			}) {
				projectV2Item {
					id
				}
			}
		}
	`, projectID, itemID, statusFieldID, statusOptionID)

	_, err = runGraphQL(mutation)
	if err != nil {
		return fmt.Errorf("failed to update item status: %w", err)
	}

	return nil
}

func escapeString(s string) string {
	s = strings.ReplaceAll(s, "\\", "\\\\")
	s = strings.ReplaceAll(s, "\"", "\\\"")
	s = strings.ReplaceAll(s, "\n", "\\n")
	return s
}

// IssueComment represents a comment on a GitHub issue
type IssueComment struct {
	ID        int    `json:"id"`
	Body      string `json:"body"`
	CreatedAt string `json:"created_at"`
	User      struct {
		Login string `json:"login"`
	} `json:"user"`
}

// GetIssueComments fetches all comments for a GitHub issue
func GetIssueComments(owner, repo string, issueNumber int) ([]IssueComment, error) {
	cmd := exec.Command("gh", "api",
		fmt.Sprintf("/repos/%s/%s/issues/%d/comments", owner, repo, issueNumber),
		"--jq", ".")

	output, err := cmd.Output()
	if err != nil {
		return nil, fmt.Errorf("failed to get issue comments: %w", err)
	}

	var comments []IssueComment
	if err := json.Unmarshal(output, &comments); err != nil {
		return nil, fmt.Errorf("failed to parse issue comments: %w", err)
	}

	return comments, nil
}

// CreateIssueComment creates a new comment on a GitHub issue
func CreateIssueComment(owner, repo string, issueNumber int, body string) error {
	// Create a JSON payload
	payload := map[string]string{
		"body": body,
	}

	payloadBytes, err := json.Marshal(payload)
	if err != nil {
		return fmt.Errorf("failed to marshal comment body: %w", err)
	}

	cmd := exec.Command("gh", "api",
		fmt.Sprintf("/repos/%s/%s/issues/%d/comments", owner, repo, issueNumber),
		"--method", "POST",
		"--input", "-")

	cmd.Stdin = bytes.NewReader(payloadBytes)

	var stderr bytes.Buffer
	cmd.Stderr = &stderr

	if err := cmd.Run(); err != nil {
		return fmt.Errorf("failed to create issue comment: %s", stderr.String())
	}

	return nil
}
